pylint/checkers/classes/class_checker.py

```diff
@@ -1506,9 +1501,8 @@ def _check_redefined_slots(
         }
 
         # Slots which are common to `node` and its parent classes
-        redefined_slots = ancestors_slots_names.intersection(slots_names)
 
-        if redefined_slots:
+        if redefined_slots := ancestors_slots_names.intersection(slots_names):
```

Comment:
This is equally hard too read

Reply:
Sounds like you don't like the language feature then? This is straightforward.